SAML2/XML.hs

```diff
@@ -95,19 +98,50 @@ type IdentifiedURI = Identified URI
 instance Identifiable URI a => XP.XmlPickler (Identified URI a) where
   xpickle = xpIdentified XS.xpAnyURI
 
+
 samlToDoc :: XP.XmlPickler a => a -> HXT.XmlTree
 samlToDoc = head
   . HXT.runLA (HXT.processChildren $ HXT.cleanupNamespaces HXT.collectPrefixUriPairs)
   . XP.pickleDoc XP.xpickle
 
+samlToDoc' :: XP.XmlPickler a => a -> HXT.XmlTree
+samlToDoc' = head . getChildren . head
+  . HXT.runLA (HXT.processChildren $ HXT.cleanupNamespaces HXT.collectPrefixUriPairs)
+  . XP.pickleDoc XP.xpickle
+
+-- | see 'docToXML''
 docToXML :: HXT.XmlTree -> BSL.ByteString
-docToXML = xshow' cquot aquot (:) . getChildren where (cquot, aquot) = escapeXmlRefs
+docToXML =  BSL.concat . HXT.runLA (HXT.xshowBlob HXT.getChildren)
```

Comment:
did this change upstream and we didn't pull the change?  how does it change behavior?  are there tests?  then we could watch both implemnetations.